cpp/src/io/functions.cpp

```diff
@@ -517,6 +517,26 @@ table_input_metadata::table_input_metadata(table_view const& table)
     table.begin(), table.end(), std::back_inserter(this->column_metadata), get_children);
 }
 
+table_input_metadata::table_input_metadata(table_metadata const& metadata)
+{
+  auto const& names = metadata.schema_info;
+
+  // Create a metadata hierarchy with naming and nullability using `table_and_metadata`
+  std::function<column_in_metadata(column_name_info const&)> get_children =
```

Comment:
Very nit picking here, but `get_children` doesn't just get children. It is also setting nullability on the metadata. Maybe something like `process_node` or `set_nullability`?

Reply:
Cut and paste strikes again :sweat_smile: I like `process_node`